tools/database/dbc/item_effect.go

```diff
@@ -33,3 +40,193 @@ func (e *ItemEffect) ToMap() map[string]interface{} {
 func GetItemEffect(effectId int) ItemEffect {
 	return dbcInstance.ItemEffects[effectId]
 }
+
+var emptyStats = stats.Stats{}
+
+func (e *ItemEffect) ToProto(itemLevel int, levelState proto.ItemLevelState) *proto.ItemEffect {
+	pe := newProtoShell(e)
+
+	statsSpellID, ilvlRppmMod := applyTrigger(e, pe, itemLevel)
+
+	pe.ScalingOptions[int32(levelState)] = buildScalingProps(statsSpellID, itemLevel)
+	if ilvlRppmMod != 0.0 && ilvlRppmMod != 1.0 {
+		pe.ScalingOptions[int32(levelState)].RppmIlvlModifier = ilvlRppmMod
+	}
+	return pe
+}
```

Comment:
Don't you need to mirror the logic used for items, where you determine all of the relevant `ItemLevelState` settings that the effect could have and then populate `ScalingOptions` entries for each of those keys?